core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -1001,6 +1001,12 @@ class KafkaConfigTest {
     assertEquals(expectedVoters, raftConfig.quorumVoterConnections())
   }
 
+  @Test
+  def testNonRoutableAddressSpecException(): Unit = {
+    assertThrows(classOf[IllegalArgumentException],
+      () => new InetAddressSpec(new InetSocketAddress("0.0.0.0", 0)))
```

Comment:
nit: I still think `RaftConfigTest` is a better home for this. It doesn't involve `KafkaConfig` at all. We could also just delete the test since it probably isn't buying us much.

Reply:
Yea, the only thing this is testing is that the `InetAddressSpec` never accepts a `0.0.0.0:0` as a parameter. We could remove it instead.